feat(eip): add resource global internet bandwidth (#4079)
huaweicloud/terraform-provider-huaweicloud: docs/resources/global_internet_bandwidth.md

---
subcategory: "Elastic IP (EIP)"
---

# huaweicloud_global_internet_bandwidth

> Manages a global internet shared bandwidth resource within HuaweiCloud.

Reviewer: `shared` should be deleted, please update them all.
```suggestion
Manages a global internet bandwidth resource within HuaweiCloud.
```

## Example Usage

```hcl
variable "enterprise_project_id" {}
variable "bandwidth_name" {}
variable "access_site" {}
variable "isp" {}
variable "bandwidth_type" {}

resource "huaweicloud_global_internet_bandwidth" "test" {
  access_site           = var.access_site
  charge_mode           = "95peak_guar"
  enterprise_project_id = var.enterprise_project_id
  size                  = 300
  isp                   = var.isp
  name                  = var.bandwidth_name
  type                  = var.bandwidth_type

  tags = {
    foo = "bar"
    key = "value"
  }
}
```

## Argument Reference

The following arguments are supported:

* `access_site` - (Required, String, ForceNew) Specifies the access site name.

>   Please refer to `data.hauweicloud_global_eip_pools` to get available `access_site` in your account.

Reviewer: suggest to use `data.hauweicloud_global_eip_pools` in the example above.

  Changing this creates a new resource.

> * `charge_mode` - (Required, String) Specifies the charge mode, only support **95peak_guar** now.

Reviewer: we can not sure if it only supports **95peak_guar**, suggest to writing like this:
 ```suggestion
* `charge_mode` - (Required, String) Specifies the charge mode, for example, **95peak_guar**.
```

* `isp` - (Required, String, ForceNew) Specifies the internet service provider of the global internet shared bandwidth.
  Please refer to `data.hauweicloud_global_eip_pools` to get available `isp` in your account.
  Changing this creates a new resource.

* `size` - (Required, Int) Specifies the size of the global internet shared bandwidth.
  The value ranges from **300 Mbit/s** to **5000 Mbit/s** in normal.

* `ingress_size` - (Optional, Int) Specifies the ingress size of the global internet shared bandwidth.
  It's **not** used for charge mode **95peak_guar**.

* `type` - (Optional, String) Specifies the type of the global internet shared bandwidth.
  Please refer to `data.hauweicloud_global_eip_pools` through `allowed_bandwidth_types[*].type` to get available `type`
  in your account.

* `tags` - (Optional, Map) Specifies the tags of the global internet shared bandwidth.

>   Changing this creates a new resource.

Reviewer: this line should be deleted.
